lib/Model/Pass/PurgeUnnamedAndUnreachableTypes.cpp

```diff
@@ -20,10 +26,74 @@ static RegisterModelPass R("purge-unnamed-and-unreachable-types",
 
 static RegisterModelPass R2("purge-unreachable-types",
                             "Remove all the types that cannot be reached from "
-                            "a type \"outside\" the type "
-                            "system itself",
+                            "a type \"outside\" the type system itself",
                             model::purgeUnreachableTypes);
 
+// Helper for fixing array constness.
+static void fixConstArrays(model::Type &Type) {
+  if (auto *Array = llvm::dyn_cast<model::ArrayType>(&Type)) {
+    if (Array->IsConst()) {
+      Array->ElementType()->IsConst() = true;
+      Array->IsConst() = false;
+    }
+
+    if (not Array->ElementType().empty())
+      fixConstArrays(*Array->ElementType());
```

Comment:
🚔  Illegal unbounded recursion on user input

Reply:
Ping

Reply:
This was solved as a part of the bigger effort to make all the recursive things into coroutines after the relevant discussion with `@fez`.